src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/transform.py

```diff
@@ -29,15 +29,46 @@ def create_transform(client, account_name, resource_group_name,
     return client.create_or_update(resource_group_name, account_name, transform_name, outputs, description)
 
 
-def add_transform_output(client, account_name, resource_group_name, transform_name, presets):
+def add_transform_output(client, account_name, resource_group_name, transform_name, preset,
+                         audio_insights_only=False, audio_language=None, on_error=None,
+                         relative_priority=None):
+    from azure.mgmt.media.models import (OnErrorType, Priority)
+
     transform = client.get(resource_group_name, account_name, transform_name)
 
-    for preset in presets:
-        transform.outputs.append(get_transform_output(preset))
+    validate_arguments(preset, audio_insights_only, audio_language)
+    transform_output = get_transform_output(preset)
+
+    if preset == 'VideoAnalyzer':
+        transform_output.preset.audio_language = audio_language
+        transform_output.preset.audio_insights_only = audio_insights_only
+    elif preset == 'AudioAnalyzer':
+        transform_output.preset.audio_language = audio_language
+
+    if on_error is not None:
```

Comment:
there is no need to neither chek if on_error is not None nor import OnErrorType ...  I think you can just do `transform_output.on_error = on_error`

Reply:
I tried doing that but it didn't work. I think it's because it doesn't allow str or OnErrorType, only OnErrorType.

Reply:
Why not just using

transform_output.on_error = OnErrorType(on_error) if on_error